src/ol_infrastructure/infrastructure/aws/opensearch/__main__.py

```diff
@@ -19,10 +19,16 @@
 search_config = pulumi.Config("opensearch")
 env_config = pulumi.Config("environment")
 stack_info = parse_stack()
+
+if stack_info.env_prefix == "open":
```

Comment:
Maybe make this more flexible and support a config parameter of `stack_prefix` or something like that as an override of the explicit stack namespace?